### DISCUSSION TITLE
Importing only required language packs in Vite

### DISCUSSION THREAD

**Discussion by maggie44** [quasarframework/quasar#12848] [General - Components / Directives / etc] [answered]
In Webpack, I would use the following to remove unused languages from the bundle. Is there an equivalent now I am moving to Vite?

```
      import(
        // https://quasar.dev/options/quasar-language-packs#dynamical-non-ssr-
        /* webpackInclude: /(en-US|de|fr|it|nb-NO|pt-BR)\.js$/ */
        `quasar/lang/${val}`
      )
```

https://quasar.dev/quasar-cli-webpack/lazy-loading
https://quasar.dev/options/quasar-language-packs#dynamical-non-ssr-

Also noting the following error on the build, which makes me think this needs reformatting:

```
[plugin:rollup-plugin-dynamic-import-variables] invalid import "import(`quasar/lang/${val}`)". Variable bare imports are not supported, imports must start with ./ in the static part of the import. For example: import(`./foo/${bar}.js`).
```

When I change to `./quasar/lang/${val}` I get:

```
[plugin:rollup-plugin-dynamic-import-variables] invalid import "import(`./quasar/lang/${val}`)". A file extension must be included in the static part of the import. For example: import(`./foo/${bar}.js`).
```

**Comment by maggie44** [comment]
`import(`./quasar/lang/${val}/index.js`)` seems to silence the errors. But not sure if it is benefiting from only importing the required languages like was done with the magic comment in Webpack?

Edit:

Or maybe not:
<img width="1234" alt="Screenshot 2022-03-19 at 19 38 31" src="https://user-images.githubusercontent.com/64841595/159135919-6b81ebcd-3bea-4816-b125-0dd6de83a12e.png">

**Comment by rstoenescu** [comment, marked as answer]
Use Vite's `import.meta.glob()` (spoiler alert: works differently, you don't just replace import() with it). Won't be able to import just a subset of all Quasar language packs, but overall, your users will only download what you'll be using -- so essentially the same effect.

**Reply by maggie44** [reply]
It looks like import.meta.glob isn't compatible with node_modules directly (https://github.com/vitejs/vite/issues/5728). 

`const lang = import.meta.glob('../../node_modules/quasar/lang/*.js')` could have been in the old import method? But when using it the promise is rejected when trying to call a function from the returned object.

**Reply by maggie44** [reply]
Been playing around with it for some time, to the best of my ability. Can't seem to find a good way around it. Queried it with the person opening the issue over on Vite, and all they had come up with is to copy the files from the node_modules in to the src directory, which of course is far from ideal (https://github.com/vitejs/vite/issues/5728#issuecomment-1073864410).

Should I reopen this thread as an issue?
